src/psyclone/tests/psyir/symbols/symbol_table_test.py

```diff
@@ -1125,6 +1121,24 @@ def test_imported_symbols():
     assert len(sym_table.imported_symbols) == 2
 
 
+def test_unresolved_datasymbols():
+    ''' Tests for the unresolved_datasymbols method. '''
+    sym_table = SymbolTable()
+    sym_table.add(DataSymbol("s1", INTEGER_TYPE))
+    # Check that we get an empty list if everything is defined
+    assert sym_table.unresolved_datasymbols == []
+    # Add a symbol with a deferred interface
+    rdef = DataSymbol("r_def", INTEGER_TYPE,
+                      interface=UnresolvedInterface())
+    sym_table.add(rdef)
+    assert sym_table.unresolved_datasymbols == [rdef]
+    # Add a symbol that uses r_def for its precision
+    scalar_type = ScalarType(ScalarType.Intrinsic.REAL, rdef)
+    sym_table.add(DataSymbol("s2", scalar_type))
+    # By default we should get this precision symbol
+    assert sym_table.unresolved_datasymbols == [rdef]
```

Comment:
I don't quite follow this bit - what is this adding to the testing done at L1134?

Reply:
Nothing really. Deleted